docs/application/dotnet/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1099 @@
+
+            /// Unset the callback
+            VoiceControlManagerClient.PreRecognitionResultUpdated -= preRecognitionResultUpdated;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the all recognition result received callback invoked when voice control engine sends the all recognition result to voice control manager.
+
+    In the callback, the recognized result, recognized text, and engine message results are included. Recognized result can include more than two voice commands, if the two voice commands have the same command string and are registered by each voice control clients.
+
+    If you want to select specific command in the recognized result, you can use `SetRecognizedCommandsSelectionDelegate()` method.
+
+    ```csharp
+    /// callback
+    void allRecognitionResultReceived(object sender, AllRecognitionResultEventArgs e)
+    {
+        /// Your code
+    }
+
+    void SetUnsetAllRecognitionResultReceivedCb()
+    {
+        try
+        {
+            /// Set the callback
+            VoiceControlManagerClient.AllRecognitionResultReceived += allRecognitionResultReceived;
+
+            /// Unset the callback
+            VoiceControlManagerClient.AllRecognitionResultReceived -= allRecognitionResultReceived;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the recognition result updated callback invoked when voice control engine updates the recognition result to voice control manager.
+
+    You can get the recognized result, recognized text, and recognized command list in this callback.
+
+    If no commands are matched, the callback returns `RecognizedResult.Rejected`:
+
+    ```csharp
+    /// callback
+    void recognitionResultUpdated(object sender, RecognitionResultUpdatedEventArgs e)
+    {
+        /// Your code
+    }
+
+    void SetUnsetRecognitionResultUpdatedCb()
+    {
+        try
+        {
+            /// Set the callback
+            VoiceControlManagerClient.RecognitionResultUpdated += recognitionResultUpdated;
+
+            /// Unset the callback
+            VoiceControlManagerClient.RecognitionResultUpdated -= recognitionResultUpdated;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+-   Set the recognized commands selection callback invoked when the voice control manager select specific voice commands in all recognized result:
+
+    When utterance is recognized, all the recognized result can include more than two voice commands if these commands have the same command string that is registered by each voice control clients.
+
+    In this case, voice control manager can select specific voice commands from all recognized result to send voice control client using `SetRecognizedCommandsSelectionDelegate`.
+
+    You can select a valid command result from the recognized command selection.
```

Comment:
Just a suggestion, would it be good to add the content in line 449 to 453 after the code block (starting line 481).

@mijin85cho, please let me know your opinion.

Reply:
@surbhig12 I'm not sure with the effect of that modification.

If you're asking this because of the colon, I'd rather suggestion to change the period of line 453 to a colon and the colon of the line 447 to a period.

Other thing. For the line 451, how about changing `SetRecognizedCommandsSelectionDelegate` to `SetRecognizedCommandsSelectionDelegate()`?

Reply:
@mijin85cho, I agree with your suggestion.

Reply:
@mijin85cho @surbhig12 Done Thank you!!